### PR TITLE
Improve high-level SDK perf, consumer groups for single client, use uuid v7

### PR DESCRIPTION
**Pull request by spetz**

### CODE REVIEW

File: sdk/src/clients/producer.rs [hunk 1]
```diff
@@ -314,13 +217,9 @@ impl IggyProducer {
...
 
     pub async fn send_one(&self, message: Message) -> Result<(), IggyError> {
-        if !self.can_send.load(ORDERING) {
-            trace!("Trying to send message in {}...", self.retry_interval);
-            sleep(self.retry_interval.get_duration()).await;
-        }
-
-        self.send_immediately(&self.stream_id, &self.topic_id, vec![message], None)
-            .await
+        let mut messages = Vec::with_capacity(1);
```
Comment (numinnex) [inline sdk/src/clients/producer.rs]: let messages = vec![message]

File: sdk/src/clients/consumer.rs [hunk 2]
```diff
@@ -675,9 +793,9 @@ impl IggyConsumerBuilder {
             partition: partition_id,
             polling_strategy: PollingStrategy::next(),
             batch_size: 1000,
-            auto_commit: AutoCommit::IntervalOrAfter(
+            auto_commit: AutoCommit::IntervalOrWhen(
                 IggyDuration::from(SEC_IN_MICRO),
```
Comment (numinnex) [inline sdk/src/clients/consumer.rs]: maybe it's good idea to create certain constants that can be easily created for example IggyDuration::Second(), etc...

**Review by numinnex — CHANGES_REQUESTED** [review]

File: sdk/src/clients/producer.rs (hunk 1 above)
Comment (spetz) [reply sdk/src/clients/producer.rs]: Done

File: sdk/src/clients/consumer.rs (hunk 2 above)
Comment (spetz) [reply sdk/src/clients/consumer.rs]: Done

**Review by numinnex — APPROVED** [review]

**Pull request merged**